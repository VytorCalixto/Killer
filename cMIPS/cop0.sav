[*]
[*] GTKWave Analyzer v3.3.37 (w)1999-2012 BSI
[*] Fri Mar 20 20:27:42 2015
[*]
[dumpfile] "/home/roberto/cMIPS/v_cMIPS.vcd"
[dumpfile_mtime] "Fri Mar 20 19:54:48 2015"
[dumpfile_size] 21630696
[savefile] "/home/roberto/cMIPS/cop0.sav"
[timestart] 5829800000
[size] 1062 920
[pos] 854 0
*-26.000000 5940000000 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
[treeopen] u_core.
[treeopen] u_core.u_alu.
[sst_width] 210
[signals_width] 265
[sst_expanded] 1
[sst_vpaned_height] 291
@28
clk
@200
-    fetch
@22
u_core.pc[31:0]
u_core.instr_fetched[31:0]
@28
u_core.pcsel[1:0]
u_core.excp_pcsel[2:0]
u_core.exception_stall
@24
u_core.nullify
@29
u_core.interrupt_taken
@200
-    decode, reg fetch
@22
u_core.rf_instruction[31:0]
@24
u_core.a_rs[4:0]
u_core.a_rt[4:0]
u_core.a_rd[4:0]
u_core.exception_dec[31:0]
@200
-    exec
@24
u_core.u_alu.operation[31:0]
@22
u_core.alu_inp_a[31:0]
u_core.alu_inp_b[31:0]
u_core.result[31:0]
@28
u_core.ovfl
@200
-   COP-0
@24
u_core.exccode[4:0]
u_core.exception_num[31:0]
@22
u_core.cause[31:0]
u_core.status[31:0]
u_core.epc[31:0]
@200
-  counter
@28
u_core.irq[5:0]
@22
u_interrupt_counter.q[29:0]
@200
-    memory
@28
u_core.b_sel[3:0]
@22
u_core.data_inp[31:0]
u_core.data_out[31:0]
d_addr[31:0]
@28
cpu_d_aval
wr
@200
-    write-back
@28
u_core.wb_wreg
@24
u_core.wb_a_c[4:0]
@22
u_core.wb_cop0_val[31:0]
@28
u_core.wb_muxc[2:0]
@22
u_core.wb_c[31:0]
[pattern_trace] 1
[pattern_trace] 0
