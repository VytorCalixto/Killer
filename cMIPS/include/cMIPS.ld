SECTIONS
{
       . = 0x00000000; /* x_INST_BASE_ADDR */
       .text : { *(.text .text.*) }
       . = 0x00040000; /* x_DATA_BASE_ADDR */
       .data    : { *(.data .data.*) }
       .rodata  : { *(.rodata .rodata.*) }
       .rodata1 : { *(.rodata1) }
       .data1   : { *(.data1) }
       .sdata   : { *(.sdata .sdata.*) }
       .lit8    : { *(.lit8) }
       .lit4    : { *(.lit4) }
       .sbss    : { *(.sbss .sbss.*) *(.scommon .scommon.*) }
       .bss     : { *(.bss .bss.*) *(COMMON) }
}

