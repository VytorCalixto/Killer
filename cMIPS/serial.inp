0
11
ffffffff
7C2



