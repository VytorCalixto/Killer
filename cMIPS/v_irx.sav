[*]
[*] GTKWave Analyzer v3.3.37 (w)1999-2012 BSI
[*] Sun May 17 23:47:43 2015
[*]
[dumpfile] "/home/roberto/cMIPS/v_cMIPS.vcd"
[dumpfile_mtime] "Sun May 17 23:34:01 2015"
[dumpfile_size] 24210019
[savefile] "/home/roberto/cMIPS/v_irx.sav"
[timestart] 6435000000
[size] 1200 908
[pos] 676 118
*-29.000000 7800000000 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
[treeopen] u_simple_uart.
[sst_width] 210
[signals_width] 221
[sst_expanded] 1
[sst_vpaned_height] 266
@28
clk
@22
u_core.pc[31:0]
@200
-    decode, reg fetch
@22
u_core.rf_instruction[31:0]
u_core.regs_a[31:0]
u_core.regs_b[31:0]
@200
-    exec
@24
u_core.u_alu.operation[31:0]
@22
u_core.alu_inp_a[31:0]
u_core.alu_inp_b[31:0]
u_core.result[31:0]
@200
-  COP0
@28
u_core.interrupt_taken
@22
u_core.epc[31:0]
u_core.cause[31:0]
u_core.status[31:0]
@200
-    memory
@28
u_core.b_sel[3:0]
@22
d_addr[31:0]
u_core.data_inp[31:0]
u_core.data_out[31:0]
@28
cpu_d_aval
u_core.mm_wrmem
@200
-    write-back
@28
u_core.wb_muxc[2:0]
u_core.wb_wreg
@22
u_core.wb_a_c[4:0]
u_core.wb_c[31:0]
@200
-  UART
@28
u_simple_uart.u_uart.s_stat
@22
u_simple_uart.u_uart.status[7:0]
@28
u_simple_uart.u_uart.s_ctrl
@22
u_simple_uart.u_uart.ctrl[7:0]
@28
u_simple_uart.u_uart.s_rx
@22
u_simple_uart.u_uart.rxreg[7:0]
@28
u_simple_uart.irq
@200
-  reception
@24
u_simple_uart.u_uart.rxcpu_dbg_st[31:0]
@28
u_simple_uart.u_uart.rx_bfr_full
@24
u_simple_uart.u_uart.rx_dbg_st[31:0]
@28
u_simple_uart.u_uart.rxclk
u_simple_uart.u_uart.rxdat
@200
-  REMOTE (fake) UART
@24
u_uart_remota.tx_dbg_st[31:0]
@28
u_uart_remota.outdat
[pattern_trace] 1
[pattern_trace] 0
