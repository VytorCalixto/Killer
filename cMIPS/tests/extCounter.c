// Testing the external counter is difficult because it counts clock cycles
// rather than instructions -- if the io/instruction latencies change then
// the simulation output also changes and comparisons become impossible.

#include "cMIPS.h"

#define N 4
#define CNT_VALUE 0x40000040    // set count to 64 cycles

void main(void) {
  int i;
  volatile int *counter;        // address of counter

  counter = (int *)IO_COUNT_ADDR;

  *counter = CNT_VALUE;

  for (i=0; i < N; i++) {       // repeat N rounds
    print(i);                   // print number of round
    *counter = (int)(CNT_VALUE + (i<<2));  // num cycles increases with i
    do {
      print((int)*counter);     // print out count value
    } while ( (*counter & 0x3fffffff) < (CNT_VALUE & 0x3ffffff) );    // done?
    to_stdout('\n');
  }

}
