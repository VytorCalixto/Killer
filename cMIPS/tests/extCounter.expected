00000000
40000001
4000000e
4000001b
40000028
40000035
40000040

00000001
40000001
4000000e
4000001b
40000028
40000035
40000042

00000002
40000001
4000000e
4000001b
40000028
40000035
40000042

00000003
40000001
4000000e
4000001b
40000028
40000035
40000042

